### DISCUSSION TITLE
copilot.vim doesn't always dismiss suggestion when exiting insert mode

### DISCUSSION THREAD

**Discussion by redbmk** [community/community#77719] [Copilot Conversations]
### Select Topic Area

Bug

### Body

I've noticed if you use `<esc>` to exit insert mode, the current suggestion will disappear. If you however use `<c-c>` or a mapping like `jk`, it will remain visible.

I most often use `jj` or `jk` to exit insert mode and, until now, if I see a dangling suggestion I've been doing `i<esc>` to clear it.

The latest workaround I found is to add `<plug>(copilot-dismiss)` to the mappings:

```vim
inoremap jj <plug>(copilot-dismiss)<c-c>:w<cr>
inoremap jk <plug>(copilot-dismiss)<c-c>
inoremap <c-c> <plug>(copilot-dismiss)<c-c>
```

This works, but I think if I was using this config in a place where copilot wasn't installed it would end up inserting `<plug>(copilot-dismiss)` as text before exiting, so it feels a little fragile.

I'm not sure why it works with `<esc>` but not other ways of going back to normal mode, but it would be awesome if the plugin could somehow detect the mode change and clear it regardless.

**Comment by jfo** [comment]
Hi,

I looked into this as I have the same problem using `<C-C>` to leave insert mode.

It's looks like the [InsertLeave](https://github.com/github/copilot.vim/blob/5b19fb001d7f31c4c7c5556d7a97b243bd29f45f/plugin/copilot.vim#L55) command is explicitly _not_ [triggered when using `<C-C>`](https://neovim.io/doc/user/insert.html#i_CTRL-C)

I have no idea why this is but it seems historically intentional as it's present in [og vim](https://vimdoc.sourceforge.net/htmldoc/autocmd.html#InsertLeave) as well as neovim.

All that said, why not use 

```
inoremap jj <esc>
inoremap jk <esc>
```

which will fix your issue, but not mine since I type `<c-c>` directly every time 🤣 .

I do wonder what the historical reasoning behind the trigger being skipped in this case is and if there may be a better candidate to clear the suggestion, but for the moment it looks like this behavior is some manner of "intentional" based on the specifications.

**Reply by redbmk** [reply]
hah such an obvious fix in hindsight, yeah using `<esc>` in the map fixes it for me. That is interesting about `<c-c>` not calling `insertLeave`. I guess it's akin to using SIGKILL instead of SIGINT? I wonder if you could remap `<c-c>` to `<esc>`...

OK actually this is weird. For me if I use `<c-c>` directly, it actually _does_ go away. If I use `inoremap jk <c-c>` however, it doesn't (but mapping it to `<esc>` does).

Makes me wonder if I do have `inoremap <c-c> <esc>` somewhere, but I'm not seeing it.

**Reply by redbmk** [reply]
Oh... it's because I did have this:

```
inoremap <c-c> <plug>(copilot-dismiss)<c-c>
```

It does look like this works:

```
inoremap <c-c> <esc>
```

but who knows what unintended side effects that could have.
